invokeai/backend/image_util/fast_guided_filter/fast_guided_filter.py

```diff
@@ -0,0 +1,283 @@
+# ruff: noqa: E741
```

Comment:
This file appears to have been copied from somewhere. Can you link to the original?

Also, why use this implementation over `cv2.ximgproc.guidedFilter`?

Reply:
The PR was made to support this new Tile model: https://huggingface.co/xinsir/controlnet-tile-sdxl-1.0

This one uses a custom version of the `guidedFilter` which they claim to be faster. I have not done benchmarks to test this myself. Functionally using the inbuilt `guidedFilter` from `cv2` should also work just fine imo.

Reply:
Seems like it would be easy to test for both speed and behavior. I think that's worth doing so we can decide whether it's worth bringing in this file.

If we *do* decide that this file is better than the cv2 version, then we should:
- [ ] Explain why we are using this over cv2 at the top of this file.
- [ ] Permalink to the source that this file was copied from.
- [ ] Remove `import cv2.ximgproc` from the invocation file.